build.sbt

```diff
@@ -8,13 +8,38 @@ scalaVersion := "2.13.0"
-  "-Ywarn-dead-code",      // Fail when dead code is present. Prevents accidentally unreachable code.
-  "-Ywarn-value-discard"   // Prevent accidental discarding of results in unit functions.
+  "-unchecked",                        // Show details of unchecked warnings.
+  "-deprecation",                      // Show details of deprecation warnings.
+  "-encoding", "UTF-8",                // Set correct encoding for Scaladoc.
+  "-feature",                          // Show details of feature warnings.
+  "-explaintypes",                     // Explain type errors in more detail.
+  "-Xcheckinit",                       // Wrap field accessors to throw an exception on uninitialized access.
+  "-Xlint",                            // Ensure best practices are being followed.
+  "-Xlint:adapted-args",               // Warn if an argument list is modified to match the receiver.
+  "-Xlint:constant",                   // Evaluation of a constant arithmetic expression results in an error.
+  "-Xlint:delayedinit-select",         // Selecting member of DelayedInit.
+  "-Xlint:doc-detached",               // A Scaladoc comment appears to be detached from its element.
+  "-Xlint:inaccessible",               // Warn about inaccessible types in method signatures.
+  "-Xlint:infer-any",                  // Warn when a type argument is inferred to be `Any`.
+  "-Xlint:missing-interpolator",       // A string literal appears to be missing an interpolator id.
+  "-Xlint:nullary-override",           // Warn when non-nullary `def f()' overrides nullary `def f'.
+  "-Xlint:nullary-unit",               // Warn when nullary methods return Unit.
+  "-Xlint:option-implicit",            // Option.apply used implicit view.
+  "-Xlint:package-object-classes",     // Class or object defined in package object.
+  "-Xlint:poly-implicit-overload",     // Parameterized overloaded implicit methods are not visible as view bounds.
+  "-Xlint:private-shadow",             // A private field (or class parameter) shadows a superclass field.
+  "-Xlint:stars-align",                // Pattern sequence wildcard must align with sequence component.
+  "-Xlint:type-parameter-shadow",      // A local type parameter shadows a type already in scope.
+  "-Ywarn-dead-code",                  // Fail when dead code is present. Prevents accidentally unreachable code.
+  "-Ywarn-dead-code",                  // Fail when dead code is present. Prevents accidentally unreachable code.
+  "-Ywarn-extra-implicit",             // Warn when more than one implicit parameter section is defined.
+  "-Ywarn-numeric-widen",              // Warn when numerics are widened.
+  "-Ywarn-unused:implicits",           // Warn if an implicit parameter is unused.
+  "-Ywarn-unused:locals",              // Warn if a local definition is unused.
+  // "-Ywarn-unused:params",              // Warn if a value parameter is unused.
+  // "-Ywarn-unused:patvars",             // Warn if a variable bound in a pattern is unused.
```

Comment:
* `params` can't be enabled because the compiler thinks `underlying` is unused (doesn't know about the macro expansion).
* `patvars` can't be enabled because of a quasiquote. And actually naming even unused vars in patterns help understanding the code.